On The Map/Controller/MapKitViewController.swift

```diff
@@ -0,0 +1,106 @@
+        Client.getStudentsLocation { (response, error) in
+            guard let response = response, error == nil else {
+                return
+            }
+            self.refetchLocations(response: response)
+        }
+    }
+    
+    // MARK: - MKMapViewDelegate
+    func mapView(_ mapView: MKMapView, viewFor annotation: MKAnnotation) -> MKAnnotationView? {
+        
+        guard annotation is MKPointAnnotation else { return nil }
+
+        let reuseId = "pin"
+        
+        var pinView = mapView.dequeueReusableAnnotationView(withIdentifier: reuseId) as? MKPinAnnotationView
+        
+        if pinView == nil {
+            pinView = MKPinAnnotationView(annotation: annotation, reuseIdentifier: reuseId)
+            pinView!.canShowCallout = true
+            pinView!.pinTintColor = .red
+            pinView!.rightCalloutAccessoryView = UIButton(type: .detailDisclosure)
+        }
+        else {
+            pinView!.annotation = annotation
+        }
+        return pinView
+    }
+    
+    func mapView(_ mapView: MKMapView, annotationView view: MKAnnotationView, calloutAccessoryControlTapped control: UIControl) {
+        if control == view.rightCalloutAccessoryView {
+            guard let toOpen = view.annotation?.subtitle else {
+                errorAlertMessage(title: "Error", message: "There is something wrong with the link.")
+                return
+            }
+            UIApplication.shared.open(URL(string: toOpen!)!, options: [:], completionHandler: nil)
+        }
+    }
+    
+    func pinLocatinDisplay() {
+        var annotations = [MKPointAnnotation]()
+
+        for student in self.students {
+                // Notice that the float values are being used to create CLLocationDegree values.
+                // This is a version of the Double type.
+                let lat = CLLocationDegrees(student.latitude)
+                let long = CLLocationDegrees(student.longitude)
+
+                // The lat and long are used to create a CLLocationCoordinates2D instance.
+                let coordinate = CLLocationCoordinate2D(latitude: lat, longitude: long)
+
+                // Here we create the annotation and set its coordiate, title, and subtitle properties
+                let annotation = MKPointAnnotation()
+                annotation.coordinate = coordinate
+                annotation.title = "\(student.firstName) \(student.lastName)"
+                annotation.subtitle = student.mediaURL
+
+                // Finally we place the annotation in an array of annotations.
+                annotations.append(annotation)
+            }
+
+            // When the array is complete, we add the annotations to the map.
+            self.mapView.addAnnotations(annotations)
+    }
+
+    func refetchLocations(response: StudentsLocationResponse) {
+        DispatchQueue.main.async {
+            self.students = response.results
+            self.pinLocatinDisplay()
+        }
+    }
```

Comment:
the name of this function is misleading because it doesn't fetch anything. It would be good to rename it something that reflects what it does, which appears to be handling a response

Reply:
That makes sense. I do always have a hard time with naming functions properly. :(